### PR TITLE
Improving the simple generator example

### PR DESCRIPTION
**Pull request by tobiasqueck**
Improving the simple generator example
* fix: correctly handle url params
* feature: added usage of `store` module to securely store credentials

### CODE REVIEW

File: examples/simple-generator/src/app/ui5.ts [hunk 1]
```diff
@@ -39,6 +39,9 @@ export type UI5SupportVersions = {
 
 let ui5VersionSupport: UI5SupportVersions;
 
+/**
+ *
+ */
```
Comment (unseen1980) [inline examples/simple-generator/src/app/ui5.ts:44]: Is this a todo?

File: examples/simple-generator/src/app/service.ts [hunk 2]
```diff
@@ -39,6 +48,54 @@ export async function getServiceInfo(generator: Generator): Promise<ServiceInfo>
...
+    if (storeCreds) {
+        const systemService = await getService<BackendSystem, string>({ entityName: 'system' });
+        const system = new BackendSystem({
+            name,
+            url: provider.defaults.baseURL!,
+            client: provider.defaults.params?.['sap-client'],
+            username: provider.defaults.auth?.username,
+            password: provider.defaults.auth?.password
+        });
+        await systemService.write(system);
```
Comment (unseen1980) [inline examples/simple-generator/src/app/service.ts:95]: If for some reason the ```write``` fails, will the store return an error or should we handle it here as well?

Comment (tobiasqueck) [reply examples/simple-generator/src/app/service.ts]: I am not fully sure to be honest, however, you make a very good point here, we need to handle the possibility that something goes wrong. Storing the credentials is not critical for the generation process, therefore, I wrapped it in a `trycatch` and log the error and continue. https://github.com/SAP/open-ux-tools/pull/649/commits/7518ecc7b8cdfcf579312c8da1734ef823abcf73
Ok for you?

File: examples/simple-generator/src/app/ui5.ts (hunk 1 above)
Comment (tobiasqueck) [reply examples/simple-generator/src/app/ui5.ts:45]: https://github.com/SAP/open-ux-tools/pull/649/commits/0931d94d9973bbdcac725c10ac85a80dff8fb61a

File: examples/simple-generator/src/app/service.ts (hunk 2 above)
Comment (unseen1980) [reply examples/simple-generator/src/app/service.ts]: 👍

File: examples/simple-generator/src/app/ui5.ts (hunk 1 above)
Comment (unseen1980) [reply examples/simple-generator/src/app/ui5.ts:45]: Thanks!

**Review by unseen1980 — APPROVED** [review]
I have tested locally but the code looks clean and easy to follow. Thanks, Tobias.

**Review by ullasholla — CHANGES_REQUESTED** [review]
Thanks @tobiasqueck, looks good.
There's one minor change you might want to consider. Cheers

**Review by ullasholla — APPROVED** [review]
Thanks @tobiasqueck, looks good to me

**Comment by tobiasqueck** [comment]
@unseen1980 @ullasholla could you please re-review?

**Review by ullasholla — APPROVED** [review]
Looks good, thanks

**Review by unseen1980 — APPROVED** [review]
👍

**Pull request merged**